packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,193 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+   */
+  readonly prefixPathRewrite?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines HTTP gateway route matching based on the URL path of the request.
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   * If path rewrite is specified, the provided `path must also end with the `/` character.
+   *
+   * @param path the exact path to match on
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static exactly(path: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ exact: path }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static regex(regex: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex: regex }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefixPathMatch the value to use to match the beginning of the path part of the URL of the request.
+   *   It must start with the '/' character. If provided as "/", matches all requests.
+   *   For example, if your virtual service name is "my-service.local"
+   *   and you want the route to match requests to "my-service.local/metrics", your prefix should be "/metrics".
+   * @param prefixRewrite Specify either disabling automatic rewrite to '/' or rewriting to specified prefix path.
+   */
+  public static startsWith(prefixPathMatch: string, prefixRewrite?: HttpGatewayRoutePathRewrite): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePrefixPathMatch(prefixPathMatch, prefixRewrite);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly pathRewrite?: HttpGatewayRoutePathRewrite,
+  ) {
+    super();
+
+    if (this.pathRewrite) {
+      if (this.prefixPathMatch[0] !== '/') {
+        throw new Error('When prefix path for the rewrite is specified, prefix path for the match must start with \'/\', '
+          + `got: ${this.prefixPathMatch}`);
+      }
+      if (this.prefixPathMatch[this.prefixPathMatch.length-1] !== '/') {
+        throw new Error('When prefix path for the rewrite is specified, prefix path for the match must end with \'/\', '
+          + `got: ${this.prefixPathMatch}`);
+      }
+    }
```

Comment:
@skinny85 - I am receiving the suggestion to remove a validation from CDK and let backend service handle the entire validation. Motivation is to reduce the churn to update in multiple tools when the validation logic is updated.

What do you think and do we have best practices on how to incorporate the validation that we have on CloudFormation?

Reply:
My answer is, as always, "it depends".

If you anticipate this logic changing a lot, remove it from the CDK.

If you don't see this logic changing in the foreseeable future, leave it in the CDK.

Reply:
Thanks Adam! Will remove the validation from CDK and then "re-request".